src/geom/elem.C

```diff
@@ -602,34 +606,34 @@ void Elem::find_point_neighbors(const Point & p,
                 {
                   if (current_neighbor->active())                // ... if it is active
                     {
-                      if (current_neighbor->contains_point(p))   // ... and touches p
+                      auto it = neighbor_set.lower_bound(current_neighbor);
```

Comment:
lower_bound can't be right here, can it?  There's no sensible ordering on a ```set<T*>```.  Surely we want find?

Reply:
Wait, I see, you're not directly using it, you're setting it up for emplace_hint later?

Reply:
Because we don't want to redundantly call contains_point() on an element that's already in the neighbor set, but since we have to hunt through the neighbor set to check that we might as well save the results of that hunt to speed up the insert later.  Sorry, I think I get it now.

Reply:
Correct. If we know from a search that we don't have it... why not use the failed search for the correct insertion which is guaranteed to happen?

Reply:
Yes - that was my thought.

Surely for most cases the set search is cheaper than contains_point().